### PR TITLE
Custom fee

### PR DESCRIPTION
**Pull request by JanKuczma**
- set fees in the contract constructor
- owner can change fees
- rename "admin_fee" to "protocol_fee"

### CODE REVIEW

File: amm/contracts/stable_pool/lib.rs [hunk 1]
```diff
@@ -783,6 +788,18 @@ pub mod stable_pool {
             Ok(())
         }
 
+        #[ink(message)]
+        fn set_fee(
+            &mut self,
+            trade_fee_bps: u16,
+            protocol_fee_bps: u16,
```
Comment (deuszx) [inline amm/contracts/stable_pool/lib.rs:795]: It'd be good to have a view methods to get current fees.

File: amm/drink-tests/src/benchmark.rs [hunk 2]
```diff
@@ -139,8 +145,8 @@ fn benchmark_2pool_burn_liquidity_imbalanced(&mut session: Session) {
             BOB,
             u128::MAX,
             vec![
-                ((INIT_SUPPLY - 1) * ONE_ICE) / (RUNS as u128),
-                (INIT_SUPPLY - ((i as u128) * imbalance)) * ONE_WOOD / (RUNS as u128),
+                INIT_SUPPLY * ONE_ICE / (RUNS as u128 + 1),
+                (INIT_SUPPLY - ((i as u128) * imbalance)) * ONE_WOOD / (RUNS as u128 + 1),
```
Comment (deuszx) [inline amm/drink-tests/src/benchmark.rs:149]: What happened here? :)

File: amm/drink-tests/src/rated_swap_tests.rs [hunk 3]
```diff
@@ -20,7 +20,10 @@ const WAZERO_DEC: u8 = 12;
 const ONE_SAZERO: u128 = 10u128.pow(SAZERO_DEC as u32);
 const ONE_AZERO: u128 = 10u128.pow(WAZERO_DEC as u32);
 
-const INIT_SUPPLY: u128 = 1_000_000; // 1M
+const INIT_SUPPLY: u128 = 1_000_000; // 100k
```
Comment (deuszx) [inline amm/drink-tests/src/rated_swap_tests.rs:23]: Looks like 1M after all :)

**Review by deuszx — COMMENTED** [review]
Why did you have to play with some of the tests? Did they break in this PR?

File: amm/drink-tests/src/benchmark.rs (hunk 2 above)
Comment (JanKuczma) [reply amm/drink-tests/src/benchmark.rs:149]: I've must have been trying something out... I'll revert this change.

**Comment by JanKuczma** [comment]
> Why did you have to play with some of the tests? Did they break in this PR?

The change in `benchmarks.rs` was unintentional. The unfinished `rated_swap_tests.rs` has been failing silently due to previous changes. Next PR will bring more robust tests to check everything throughout.

**Review by deuszx — APPROVED** [review]

**Pull request merged**